src/SarifCli/Rules/UseAbsolutePathsForNestedFileUriFragments.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.CodeAnalysis.Sarif.Cli.Rules
+{
+    public class UseAbsolutePathsForNestedFileUriFragments : SarifValidationSkimmerBase
+    {
+        public override string FullDescription => RuleResources.SV0002_UseAbsolutePathsForNestedFileUriFragmentsDescription;
+
+        /// <summary>
+        /// SV0002
+        /// </summary>
+        public override string Id => RuleId.UseAbsolutePathsForNestedFileUriFragments;
+
+        protected override IEnumerable<string> FormatIds
+        {
+            get
+            {
+                return new string[]
+                {
+                    nameof(RuleResources.SV0002_DefaultFormatId)
+                };
+            }
+        }
+
+        protected override void Analyze(FileData fileData, string fileKey, string filePointer)
+        {
+            try
+            {
+                Uri fileUri = new Uri(fileKey);
+                if (UriHasNonAbsoluteFragment(fileUri))
+                {
+                    LogResult(
+                        ResultLevel.Warning,
+                        filePointer,
+                        nameof(RuleResources.SV0002_DefaultFormatId),
+                        fileUri.OriginalString);
+                }
+            }
+            catch
```

Comment:
>catch [](start = 12, length = 5)

please get rid of this general catch handler. i.e., discover specific exceptions for malformed URIs and let all other exceptions through